src/index.rs

```diff
@@ -28,10 +28,36 @@ const RELOAD_POLICY: &str = "commit";
 /// on the index object.
 #[pyclass]
 pub(crate) struct IndexWriter {
-    inner_index_writer: tv::IndexWriter,
+    inner_index_writer: Option<tv::IndexWriter>,
     schema: tv::schema::Schema,
 }
 
+impl IndexWriter {
+    fn inner(&self) -> PyResult<&tv::IndexWriter> {
+        self.inner_index_writer.as_ref().ok_or_else(|| {
+            exceptions::PyValueError::new_err(
```

Comment:
I do wonder whether it makes sense to turn this into a `ValueError` since this happening would be a logic bug in the calling Python code, i.e. a `PanicException` seems appropriate?

Reply:
I'd have gone for something more python-native like [RuntimeError](https://docs.python.org/3/library/exceptions.html#RuntimeError) myself, but no strong feelings about this.